include/seqan3/alignment/pairwise/alignment_algorithm.hpp

```diff
@@ -301,26 +370,41 @@ class alignment_algorithm :
     template <typename first_range_t,
               typename second_range_t,
               typename cache_t>
-    void compute_matrix(first_range_t const & first_range,
-                        second_range_t const & second_range,
+    void compute_matrix(first_range_t & first_range,
+                        second_range_t & second_range,
                         cache_t & cache)
     {
         using std::get;
         auto const & score_scheme = get<align_cfg::scoring>(*cfg_ptr).value;
         ranges::for_each(first_range, [&, this](auto seq1_value)
         {
+            // Move internal matrix to next column.
+            this->next_column();
+
             auto col = this->current_column();
-            this->init_row_cell(*seqan3::begin(col), cache);
+            this->init_row_cell(*std::ranges::begin(col), cache);
 
             auto second_range_it = std::ranges::begin(second_range);
-            ranges::for_each(col | ranges::view::drop_exactly(1), [&, this] (auto & cell)
+            ranges::for_each(col | ranges::view::drop_exactly(1), [&, this] (auto && cell)
```

Comment:
in which case does a column generate items on access?

Reply:
Always. it is internally a zip_view.